src/index.ts

```diff
@@ -1,62 +1,61 @@
 import { interfaces } from 'inversify';
-import { ComponentOptions } from 'vue/types/options';
 import { PluginFunction } from 'vue/types/plugin';
-import { Vue, VueConstructor } from 'vue/types/vue';
+import { Vue } from 'vue/types/vue';
+import { createDecorator } from 'vue-class-component'
 
 declare const Reflect: {
     getMetadata(metadataKey: string, target: unknown, propertyKey: string): unknown;
 };
 
 declare module "vue/types/options" {
     interface ComponentOptions<V extends Vue> {
-            container?: interfaces.Container;
-            dependencies?: { [key: string]: interfaces.ServiceIdentifier<unknown> };
+        container?: interfaces.Container;
+        dependencies?: { [key: string]: interfaces.ServiceIdentifier<unknown> };
     }
 }
 
-type InjectableVueClass = VueConstructor<Vue> & {
-    options: ComponentOptions<Vue>
-};
-
 const VueInversify: PluginFunction<unknown> = instance => {
     instance.mixin({
         beforeCreate() {
             const dependencies = this.$options.dependencies;
-            const container = this.$options.container || (this.$options.container = getContainer(this));
             
             if (dependencies) {
+                const container = this.$options.container || (this.$options.container = getContainer(this));
+
                 if (!container)
                     throw new TypeError('vue-inversify: encountered props marked with @inject but no container was provided when creating the Vue instance');
-
+                
                 Object.keys(dependencies).forEach(key => {
+                    if (Object.getOwnPropertyDescriptor(this, key))
```

Comment:
This not going to flip out? I thought the prop might be defined but we're overwriting it with the injected value

Reply:
no, the opposite, because the properties aren't specifying configurable they cannot be overwritten, in the event multiple versions are registered and are running redundantly (or you have somehow added a field with the same name through some other process) this prevents it overriding that and/or breaking

Reply:
I had assumed that Vue may have declared the variable as a regular `data` property prior to our process (because really it's just a decorated data property). But I guess this must run before that process then

Reply:
The field won't exist on the object if declared but unassigned, but also vue-class-component won't copy it across if it is undefined. If it not undefined this code won't override it.

Which is the 'correct' way to handle this otherwise unusual case, I am not sure.

see: https://github.com/vuejs/vue-class-component/blob/16433462b40aefecc030919623f17b0ec9afe61c/src/data.ts#L36

Reply:
correct, however I will investigate what vue-property-decorator does to help inform 'correct' behaviour